File: wignusi.UI/src/app/api/models/book-dto.ts
/* tslint:disable */
/* eslint-disable */
import { AuthorDto } from '../models/author-dto';
export interface BookDto {
  authors?: null | Array<AuthorDto>;
  authorsIds?: null | Array<number>;
  description?: null | string;
  image?: null | string;
  isAvialable?: boolean;
  price?: number;
  published?: string;
  publisher?: null | string;
  tags?: null | Array<string>;
  title?: null | string;
}

File: wignusi.BE/wignusi.Domain/Dtos/BookDto.cs
using System.ComponentModel.DataAnnotations;

namespace wignusi.Domain.Dtos
{
    public record BookDto(
        [Required] string Title,
        [Required] string Description,
        [Required] string Image,
        string Publisher,
        DateTime Published,
        [Required] decimal Price,
        [Required] bool IsAvialable,
        AuthorDto[]? authors,
        long[]? authorsIds,
        [Required] string[] tags
        );
}
